### ISSUE TITLE
lua error 10/10

### ISSUE THREAD

**Issue by ibwukster** [Hoizame/AtlasLootClassic#399]
**AtlasLoot version**
current as of 10/10/23

**Lua error**
Message: Interface/AddOns/AtlasLootClassic/Data/Companion.lua:674: table index is nil
Time: Tue Oct 10 18:38:36 2023
Count: 1
Stack: Interface/AddOns/AtlasLootClassic/Data/Companion.lua:674: table index is nil
[string "@Interface/AddOns/AtlasLootClassic/Data/Companion.lua"]:674: in function <Interface/AddOns/AtlasLootClassic/Data/Companion.lua:669>
[string "@Interface/AddOns/AtlasLootClassic/Data/Companion.lua"]:682: in function <Interface/AddOns/AtlasLootClassic/Data/Companion.lua:677>
[string "@Interface/AddOns/AtlasLootClassic/Data/Companion.lua"]:689: in function `func'
[string "@Interface/AddOns/AtlasLootClassic/AtlasLoot.lua"]:42: in function <Interface/AddOns/AtlasLootClassic/AtlasLoot.lua:34>

Locals: typ = "MOUNT"
(for index) = 1
(for limit) = 20
(for step) = 1
i = 1
creatureID = nil
(*temporary) = <table> {
 24480 = true
 194870 = true
 28513 = true
}
(*temporary) = 1
(*temporary) = "table index is nil"
COLLECTED_COMPANIONS = <table> {
 24480 = true
 194870 = true
 28513 = true
}

**Comment by jazminite** [comment]
PR #398 fixes this issue

**Comment by Anyalah** [comment]
I have created a [release](https://github.com/Anyalah/AtlasLootClassic/releases/tag/v3.4.3.pre-release) here which has a direct download link: 
[
Direct Download Link](https://github.com/Anyalah/AtlasLootClassic/releases/download/v3.4.3.pre-release/AtlasLootClassic_Phase4.zip)

**Comment by vgullotta8** [comment]
Awesome, thank you, and thank you for the work on this awesome addon!

**Comment by Devanna909** [comment]
Yes, thank you for the direct link also, couldn't figure out why it was still broke.

**Comment by kazso** [comment]
> Only issue I'm seeing is every time you hit a load screen, you get the following chat message "AtlasLoot-Def: Player "X-Server" sends version '3010003'! Anyway to fix this? Cause once you join a raid, you'll see that constantly with load screens or reloads.

Comment out this line in AtlasLoot.lua (line 182 in this pre-release version):
`AtlasLoot:DevPrint(format("Player '|cff00FAF6%s|r' sends version '|cff00FF96%d|r'!", sender, v))`

**Comment by vgullotta8** [comment]
When I open atlas loot classic, it says No Module Found in the Select Module dropdown. If I try to switch from Wrath to Classic or TBC or back to Wrath, it throws this LUA error:

Message: Interface/AddOns/AtlasLootClassic/ItemDB/ItemDB.lua:281: attempt to concatenate local 'addonName' (a nil value)
Time: Wed Oct 11 13:36:33 2023
Count: 3
Stack: Interface/AddOns/AtlasLootClassic/ItemDB/ItemDB.lua:281: attempt to concatenate local 'addonName' (a nil value)

Locals:

**Comment by ZhandPally** [comment]
> > Only issue I'm seeing is every time you hit a load screen, you get the following chat message "AtlasLoot-Def: Player "X-Server" sends version '3010003'! Anyway to fix this? Cause once you join a raid, you'll see that constantly with load screens or reloads.
> 
> Comment out this line in AtlasLoot.lua (line 182 in this pre-release version): `AtlasLoot:DevPrint(format("Player '|cff00FAF6%s|r' sends version '|cff00FF96%d|r'!", sender, v))`

Awesome, thank you!

**Comment by Iowerth** [comment]
> I have created a [release](https://github.com/Anyalah/AtlasLootClassic/releases/tag/v3.4.3.pre-release) here which has a direct download link: [ Direct Download Link](https://github.com/Anyalah/AtlasLootClassic/releases/download/v3.4.3.pre-release/AtlasLootClassic_Phase4.zip)

Thanks, but why did you delete all localizations?

**Comment by Anyalah** [comment]
@Iowerth 

> @Anyalah
> 
> > I have created a [release](https://github.com/Anyalah/AtlasLootClassic/releases/tag/v3.4.3.pre-release) here which has a direct download link: [ Direct Download Link](https://github.com/Anyalah/AtlasLootClassic/releases/download/v3.4.3.pre-release/AtlasLootClassic_Phase4.zip)
> 
> Thanks, but why did you delete all localizations?

No localization files were deleted in my commit, you can see the changes if you follow this link: https://github.com/Hoizame/AtlasLootClassic/pull/398/commits/0bc91eb6899bebbf3b99072b446c1b83778ddeb9
I may have added text that needs to be localised? Maybe you can be more specific if you're experiencing an issue.

**Comment by Iowerth** [comment]
@Anyalah 
> No localization files were deleted in my commit, you can see the changes if you follow this link: [0bc91eb](https://github.com/Hoizame/AtlasLootClassic/commit/0bc91eb6899bebbf3b99072b446c1b83778ddeb9) I may have added text that needs to be localised? Maybe you can be more specific if you're experiencing an issue.

Open your archive please and you'll see that all localization files (in `*\Locales` folders) have size 300 bytes - they are almost empty.

**Comment by Syllinger** [comment]
When does Curseforge get updated?

**Comment by Iowerth** [comment]
@Syllinger 
As soon as it possible.

**Comment by Anyalah** [comment]
> @Anyalah
> 
> > No localization files were deleted in my commit, you can see the changes if you follow this link: [0bc91eb](https://github.com/Hoizame/AtlasLootClassic/commit/0bc91eb6899bebbf3b99072b446c1b83778ddeb9) I may have added text that needs to be localised? Maybe you can be more specific if you're experiencing an issue.
> 
> Open your archive please and you'll see that all localization files (in `*\Locales` folders) have size 300 bytes - they are almost empty.

Okay I see now, there is a comment in each file saying these are translated on curseforge, it may be that it is populated when the addon is uploaded to curse? if you look at the current version on github its the same as in my commit with all those files being mostly empty. I am sure they will be in the proper release when Hoizame is back to do that.

**Comment by Gaspasser5** [comment]
After using the direct download link, Where would I extract all files to?

**Comment by Pocokk** [comment]
> After using the direct download link, Where would I extract all files to?

World of Warcraft/_classic/Interface/AddOns

**Comment by DieQuelle** [comment]
Still not fixed?

**Comment by Autobuz** [comment]
any way to disable the chat spam with "AtlasLoot-Dev: Player 'playername' sends version '3010003' whenever someone joins/leaves the group?

![image](https://github.com/Hoizame/AtlasLootClassic/assets/90157244/74613cc6-9f03-4c08-a41e-7b86a12e7da8)

**Comment by Iowerth** [comment]
@Autobuz https://github.com/Hoizame/AtlasLootClassic/issues/399#issuecomment-1758306068
